front/src/modules/timesStops/hooks/useTimesStopsTableData.ts

```diff
@@ -0,0 +1,248 @@
+  simulatedPathItemTimes?: Extract<SimulationSummary, { isValid: true }>['pathItemTimes'],
+  operationalPointsOnPath?: PathPropertiesFormatted['operationalPoints']
+): TimesStopsRowNew[] => {
+  const { t } = useTranslation('operational-studies');
+  const { getTrackSectionsByIds } = useScenarioContext();
+  const displayOnlyPathSteps = useSelector(getDisplayOnlyPathSteps);
+
+  const pathStepOps = usePathOps(infraId, selectedTrain?.path);
+
+  const trackIds = useMemo(() => {
+    const path = selectedTrain?.path || [];
+    const trackIdsInPathSteps = path.flatMap(({ location }) => [
+      ...('track' in location ? [location.track] : []),
+      ...('track_reference' in location &&
+      location.track_reference &&
+      'track_id' in location.track_reference
+        ? [location.track_reference.track_id]
+        : []),
+    ]);
+    const trackIdsOnPath = (operationalPointsOnPath || []).map((op) => op.part.track);
+    return [...trackIdsInPathSteps, ...trackIdsOnPath];
+  }, [selectedTrain?.path, operationalPointsOnPath]);
+
+  const [trackSections, setTrackSections] = useState<Record<string, TrackSection>>({});
+  useEffect(() => {
+    const fetchTrackSections = async () => {
+      const sections = await getTrackSectionsByIds(trackIds);
+      setTrackSections(sections);
+    };
+    fetchTrackSections();
+  }, [trackIds]);
+
+  const startDatetime = useMemo(
+    () => selectedTrain && new Date(selectedTrain.start_time),
+    [selectedTrain]
+  );
+
+  const scheduleByAt = useMemo(
+    () => selectedTrain && keyBy(selectedTrain.schedule, 'at'),
+    [selectedTrain]
+  );
+
+  const rows = useMemo(() => {
+    if (!selectedTrain || !startDatetime || !scheduleByAt) return [];
+
+    const pathStepRowsById = new Map(
+      selectedTrain.path.map((pathStep, stepIndex) => {
+        const matchingOp = pathStepOps.find((op) => matchOpRefAndOp(pathStep.location, op));
+
+        const name = getOperationalPointName(
+          matchingOp,
+          pathStep.location,
+          stepIndex,
+          selectedTrain.path.length,
+          t
+        );
+        const trackName =
+          'track' in pathStep.location
+            ? (trackSections[pathStep.location.track]?.extensions?.sncf?.track_name ??
+              pathStep.location.track)
+            : getTrackReferenceLabel(trackSections, pathStep.location.track_reference);
+
+        const schedule = scheduleByAt[pathStep.id];
+        const computedArrivalMs = simulatedPathItemTimes?.final[stepIndex];
+
+        // TODO: Add margin fields here when implementing margin columns
+        // (theoreticalMargin, theoreticalMarginSeconds, calculatedMargin, diffMargins)
+
+        const row = buildTableRow({
+          id: pathStep.id,
+          name,
+          ch: matchingOp?.extensions?.sncf?.ch,
+          trackName,
+          startDatetime,
+          schedule,
+          computedArrivalMs,
+        });
+
+        return [pathStep.id, row] as const;
```

Comment:
Why do you need `as const` here ?
```suggestion
        return [pathStep.id, row];
```

Reply:
I had a type inference issue that has since been fixed, forgot to remove it. Good catch!